pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/participant/ParticipantExitService.java

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.dsm.service.participant;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.api.client.http.HttpStatusCodes;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.DSMServer;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.KitDiscard;
+import org.broadinstitute.dsm.db.KitRequestShipping;
+import org.broadinstitute.dsm.db.ParticipantExit;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.exception.EntityNotFound;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.util.DDPRequestUtil;
+
+
+@Slf4j
+public class ParticipantExitService {
+
+    public Map<String, ParticipantExit> getExitedParticipants(String realm) {
+        return ParticipantExit.getExitedParticipants(realm, true);
+    }
+
+    public List<KitDiscard> exitParticipant(String realm, String ddpParticipantId, String userId, boolean inDDP) {
+        _exitParticipant(realm, ddpParticipantId, userId, inDDP);
+
+        // NOTE: The following code was lifted as-is from ParticipantExitRoute.java. I did not attempt to
+        // clean it up since we do not have functional tests in place. We will verify with our system tests. -DC
+        List<KitRequestShipping> kitRequests =
+                KitRequestShipping.getKitRequestsByParticipant(realm, ddpParticipantId, true);
+        List<KitDiscard> kitsNeedAction = new ArrayList<>();
+        log.info("Found {} kit requests", kitRequests.size());
+        Optional<DDPInstanceDto> maybeInstance = new DDPInstanceDao().getDDPInstanceByInstanceName(realm);
+        for (KitRequestShipping kit : kitRequests) {
+            if (kit.getScanDate() != 0 && kit.getReceiveDate() == 0) {
+                String discardId = KitDiscard.addKitToDiscard(kit.getDsmKitRequestId(), KitDiscard.HOLD);
+                kitsNeedAction.add(new KitDiscard(discardId, kit.getKitTypeName(), KitDiscard.HOLD));
+            } else {
+                //refund label of kits which are not sent yet
+                KitRequestShipping.refundKit(kit.getDsmKitRequestId(), DSMServer.getDDPEasypostApiKey(realm),
+                        maybeInstance.orElse(null));
+            }
+        }
+        return kitsNeedAction;
+    }
+
+    /**
```

Comment:
Should this javadoc be on the public method instead of the protected one?

Reply:
I mostly wanted to draw attention to `inDDP`, since it's use was odd.
I'll take a crack at the public method, but was hoping to avoid explaining why it returns a list of `KitDiscard` since I do not think it is used in the UI. Although I can hazard a guess that it is kits left dangling by the withdrawal.
Do you know what that's for?

Reply:
I don't, but Pegah probably knows.  Either way, l'm up for holding off on this for another time.